plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSink.java

```diff
@@ -361,7 +365,9 @@ private void closeWriter(HiveWriter writer)
             HiveWriter writer = writers.get(writerIndex);
             if (writer != null) {
                 // if current file not too big continue with the current writer
-                if (bucketFunction != null || writer.getWrittenBytes() <= targetMaxFileSize.orElse(Long.MAX_VALUE)) {
+                // for transactional tables we don't want to split output files because there is an implicit bucketing with
+                // 1 bucket so we want to stay consistent with bucketing behaviour for non transactional tables
+                if ((bucketFunction != null || isTransactional) || writer.getWrittenBytes() <= targetMaxFileSize.orElse(Long.MAX_VALUE)) {
```

Comment:
It's not about consistency between non-transactional bucketed and transactional, non-bucketed (implicitly bucketed) tables.
The naming rules for the two are different.
For transactional tables, the naming convention is `bucket_<bucket-number>` (eg bucket_00000). Doesn't contain any random, or incrementing part, so we simply cannot create more than once file.
https://github.com/trinodb/trino/blob/d79f6ea8c4fe279ae1d5715ad97680db9f57fdcb/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveWriterFactory.java#L452

For bucketed, non-transactional tables -- actually i am not sure why we have this condition here. @dain  would know.
My reading of the code leads to the following naming pattern for bucketed files
```
format("0%s_0_%s", paddedBucket, queryId.get())
```

-- if this is the right one (didn't test), then it's constant for `bucket x query`, so we cannot create more than 1 file either. (We could easily improve that by incrementing this `_0_` part, but that's another story).

The condition seems good, but comments needs rewording.

Reply:
redundant `(...)` parens in sequence of `||`

Reply:
I meant to be consistent with the behaviour -> when table is non transactional and has 1 bucket, writer count is ignored and only 1 file is created. I must have used wrong wording.